dateparser/date_parser.py

```diff
@@ -181,8 +182,11 @@ def parse(self, date_string, settings=None):
 
         if tz is not None:
             date_obj = tz.localize(date_obj)
+        elif not settings.NO_DEFAULT_TIMEZONE:
+            date_obj = pytz.timezone(settings.TIMEZONE).localize(date_obj)
 
-        date_obj = apply_timezone(date_obj, settings.TIMEZONE)
+        if settings.TIMEZONE_CONVERSION and not settings.NO_DEFAULT_TIMEZONE:
+            date_obj = apply_timezone(date_obj, settings.TIMEZONE)
 
```

Comment:
Hi @demelziraptor, thank you so much for the detailed explanation. I understand that `settings` for timezone can be improved and you took the right first step in that direction. However, I'd like to propose re-imagining settings like below:

- RETURN_AS_TIMEZONE_AWARE: (configuration's single responsibility is to attach `tzinfo` object to returned datetime.)
   - True (attaches tzinfo) (if timezone is detected it gets applied and attached)
   - False (no tzinfo) (if timezone detected it gets applied)
- TIMEZONE_CONVERSION:
  - True (converts to timezone defined by `TIMEZONE` setting)
  - False (no conversion occurs)
- TIMEZONE (to convert to a certain timezone if TIMEZONE_CONVERSION is set to True)
  - UTC # by default or whatever tz specified.

And instead of changes on line nos. 185, 186, 188 and 189, apply following changes.

```
@@ -182,7 +182,8 @@ class DateParser(object):
         if tz is not None:
             date_obj = tz.localize(date_obj)

-        date_obj = apply_timezone(date_obj, settings.TIMEZONE)
+        if settings.TIMEZONE_CONVERSION:
+            date_obj = apply_timezone(date_obj, settings.TIMEZONE)

         if not settings.RETURN_AS_TIMEZONE_AWARE:
             date_obj = date_obj.replace(tzinfo=None)
```

Proposed changes would enable returning timezones which are detected automatically if `settings={'RETURN_AS_TIMEZONE_AWARE': True, 'TIMEZONE_CONVERSION': False}`. e.g.

```
In [6]: parse(u'23 March 2000, 1:21 PM EST', settings={'RETURN_AS_TIMEZONE_AWARE': True, 'TIMEZONE_CONVERSION': False})
Out[6]: datetime.datetime(2000, 3, 23, 13, 21, tzinfo=<StaticTzInfo 'EST'>)
```

You can also verify the changes for other scenarios, configuring these settings accordingly. If it sill miss some cases, mention them here and we'll figure out together how to resolve them.

Thank you!